parser-typechecker/src/Unison/Server/Endpoints/FuzzyFind.hs

```diff
@@ -0,0 +1,191 @@
+import           Unison.Prelude
+import           Unison.Codebase                ( Codebase )
+import qualified Unison.Codebase               as Codebase
+import qualified Unison.Codebase.Path          as Path
+import qualified Unison.HashQualified'         as HQ'
+import qualified Unison.HashQualified          as HQ
+import           Unison.Parser                  ( Ann )
+import qualified Unison.Server.Backend         as Backend
+import           Unison.Server.Errors           ( backendError
+                                                , badNamespace
+                                                )
+import           Unison.Server.Types            ( mayDefault
+                                                , HashQualifiedName
+                                                , NamedTerm
+                                                , NamedType
+                                                , DefinitionDisplayResults(..)
+                                                , TypeDefinition(..)
+                                                )
+import           Unison.Util.Pretty             ( Width )
+import           Unison.Var                     ( Var )
+import qualified Unison.Codebase.ShortBranchHash
+                                               as SBH
+import qualified Data.Text                     as Text
+import qualified Text.FuzzyFind as FZF
+import qualified Unison.Codebase.Branch as Branch
+import Unison.NameSegment
+import           Unison.Server.Syntax           ( SyntaxText )
+import qualified Unison.Reference              as Reference
+import qualified Data.Map                      as Map
+import Unison.Codebase.Editor.DisplayObject
+
+type FuzzyFindAPI =
+  "find" :> QueryParam "rootBranch" SBH.ShortBranchHash
+         :> QueryParam "relativeTo" HashQualifiedName
+         :> QueryParam "limit" Int
+         :> QueryParam "renderWidth" Width
+         :> QueryParam "query" String
+         :> Get '[JSON] [(FZF.Alignment, FoundResult)]
+
+instance ToSample FZF.Alignment where
+  toSamples _ = noSamples
+
+instance ToParam (QueryParam "limit" Int) where
+  toParam _ =
+    DocQueryParam
+      "limit"
+      ["1", "10", "20"]
+      "The maximum number of results to return. Defaults to 10."
+      Normal
+
+instance ToParam (QueryParam "query" String) where
+  toParam _ =
+    DocQueryParam
+      "query"
+      ["foo", "ff", "td nr"]
+      "Space-separated subsequences to find in the name of a type or term."
+      Normal
+
+instance ToJSON FZF.Alignment where
+   toEncoding = genericToEncoding defaultOptions
+instance ToJSON FZF.Result where
+   toEncoding = genericToEncoding defaultOptions
+instance ToJSON FZF.ResultSegment where
+   toEncoding = genericToEncoding defaultOptions
+
+deriving instance ToSchema FZF.Alignment
+deriving instance ToSchema FZF.Result
+deriving instance ToSchema FZF.ResultSegment
+
+data FoundTerm = FoundTerm
+  { bestTermName :: HashQualifiedName
```

Comment:
@runarorama I noticed when integrating that we had named this one `bestTermName` and the other `bestFoundTypeName`. Should we name this one `bestFoundTermName` to match?

Reply:
good idea